proxy/gateway/nodejs/index.js

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+let ChannelFactory = require('./lib/channel_factory.js');
+let AttachedError = require('./lib/exceptions.js').AttachedError;
+
+class ProxyGateway {
+  constructor(module, channelFactory) {
+    if (module === undefined) throw new TypeError("Missing 'module' argument");
+    this.module = module;
+    this.channelFactory = channelFactory || new ChannelFactory();
+    this.controlChannel = null;
+    this.messageChannel = null;
+  }
+
+  _disconnectMessageChannel() {
+    if (this.messageChannel) {
+      this.messageChannel.disconnect();
+      this.messageChannel = null;
+    }
+  }
+
+  onMessage(msg) {
+    this.module.receive(msg);
+  }
+
+  onCreate(messageChannelId, args) {
+    this.messageChannel = this.channelFactory.createMessageChannel();
```

Comment:
Good catch. I just pushed another commit that destroys the existing connection (and module) cleanly if the Create control message comes through twice without an intervening Destroy.